en/docs/apis/advanced-topics/connect-to-an-external-identity-provider.md

```diff
@@ -0,0 +1,146 @@
+# Connect to an External Identity Provider
+
+Choreo uses an in-built Identity Provider (IdP) by default to manage OAuth clients and generate tokens required to authenticate Choreo APIs. Choreo also grants organization administrator privileges to users to configure a required authorization server as an identity provider(IdP) via the Choreo Console. This allows you to [add one or more external identity providers](#add-an-external-identity-provider) to your Choreo organization and [obtain access tokens from the identity providers](#obtain-an-access-token-from-an-external-identity-provider) to exchange and get Choreo access tokens to invoke APIs. This capability lets you expose your APIs to users who are external to Choreo.
+
+## Token exchange
+
+Choreo supports the token exchange grant type to exchange a JSON web token (JWT) issued by an external IdP for a Choreo token. The token exchange grant type uses the protocol defined in the [OAuth 2.0 token exchange specification](https://datatracker.ietf.org/doc/html/rfc8693), which describes how to request and obtain security tokens from OAuth 2.0 authorization servers and exchange a security token for another type of token. The following diagram depicts the token exchange flow in Choreo:
```

Comment:
```suggestion
Choreo supports the token exchange grant type to exchange a JSON Web Token (JWT) issued by an external IdP for a Choreo token. The token exchange grant type uses the protocol defined in the [OAuth 2.0 token exchange specification](https://datatracker.ietf.org/doc/html/rfc8693), which describes how to request and obtain security tokens from OAuth 2.0 authorization servers and exchange a security token for another type of token. The following diagram depicts the token exchange flow in Choreo:
```

Reply:
The guideline on this is "Capitalize the spelled-out version of the abbreviation only if the long form is a proper noun or is conventionally capitalized. That is, don't capitalize it only because the abbreviation includes capital letters." [1]
So +1 to go with the suggestion considering that the term "JSON Web Token" is conventionally capitalized.
[1] https://developers.google.com/style/abbreviations?hl=en#spelling-out